infra/core/search/search-services.bicep

```diff
@@ -47,7 +47,7 @@ resource search 'Microsoft.Search/searchServices@2021-04-01-preview' = {
   // The free tier does not support managed identity
   identity: searchIdentityProvider
   properties: {
-    authOptions: authOptions
+    authOptions: disableLocalAuth ? null : authOptions
```

Comment:
It causes an error if authOptions is an empty object, which is the default in this file, and it didn't seem to work to set that to null in main.bicep (perhaps it gets set to an empty object as it considers null to be a non-setting?).

I can make this change in azure-dev if this PR is approved. cc @jongio

Reply:
the default is null not an empty object right? I do not think empty object is valid according to our swagger, but null is

Reply:
The default at the top of this file is:

param authOptions object = {}

So that's what got set if I removed any specification of authOptions in main.bicep. However, the caused an error in the search service in combination with disableLocalAuth. Hence, my change here to explicitly set it to null when disableLocalAuth is set to true.

I also tried to set authOptions:null in main.bicep, to avoid having to change in Bicep, but I still got the error, so I think Bicep might see the "null" and decide that it's a falsy value that should be replaced with the default {} in this file (versus keeping it as null).

Reply:
hmm, good to know